CMakeLists.txt

```diff
@@ -48,6 +48,8 @@ if(BUILD_TUTORIALS)
 #    add_subdirectory(examples/rabbitmq_tutorials)
 endif()
 
+option(BUILD_PURE "build in pure mode" OFF)
```

Comment:
It would be a good idea to prefix `BUILD_PURE` with something unique. Like `AMQP_CPP_BUILD_PURE`.